repos/fdbt-site/src/components/CapTable.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { ReactElement } from 'react';
+import { Cap, ErrorInfo } from '../interfaces';
+import ExpirySelector from './ExpirySelector';
+import FormElementWrapper, { FormGroupWrapper } from './FormElementWrapper';
+
+interface CapTableProps {
+    numberOfCapsToDisplay: number;
+    errors: ErrorInfo[];
+    userInputtedCaps: Cap[];
+}
+
+export const renderTable = (index: number, errors: ErrorInfo[], userInputtedCaps: Cap[] = []): ReactElement => (
+    <fieldset key={index} className="govuk-fieldset">
+        <legend className="govuk-fieldset__legend govuk-visually-hidden">{`Enter details for cap ${index + 1}`}</legend>
+        <div className="flex-container">
+            <div className="govuk-!-margin-left-4 govuk-!-margin-right-2">
+                <FormGroupWrapper errors={errors} errorIds={[`cap-name-${index}`]} hideErrorBar>
+                    <>
+                        {index === 0 ? (
+                            <>
+                                <label className="govuk-label" htmlFor={`cap-name-${index}`}>
+                                    <span className="govuk-visually-hidden">{`Cap Name - Cap ${index + 1}`}</span>
+                                    <span aria-hidden>Cap name</span>
+                                </label>
+                                <span className="govuk-hint" id={`cap-name-hint-${index}`}>
+                                    50 characters max
+                                </span>{' '}
+                            </>
+                        ) : (
+                            ''
+                        )}
+
+                        <FormElementWrapper
+                            errors={errors}
+                            errorId={`cap-name-${index}`}
+                            errorClass="govuk-input--error"
+                            hideText
+                            addFormGroupError={false}
+                        >
+                            <input
+                                className="govuk-input govuk-input--width-40 govuk-product-name-input__inner__input"
+                                id={`cap-name-${index}`}
+                                name={`capNameInput${index}`}
+                                type="text"
+                                aria-describedby={`cap-name-hint-${index}`}
+                                maxLength={50}
```

Comment:
better if we can also add the minLength={2} as we are having the same constraint in the API validation

Reply:
Unsure what behaviour this will add to the UI - we don't want any client side validation to happen

Reply:
then, do we need the maxLength also ? just asking

Reply:
Realistically I don't think we need max length either, but kept it in because we have it in the other component.